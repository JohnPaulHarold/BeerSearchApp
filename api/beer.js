import fetch from 'node-fetch';
export const BASE_URL = 'https://api.punkapi.com/v2';

const returnJson = res => res.json();
const returnBody = json => json;

export const searchBeersByName = term => {
  const fixedTerm = term.replace(/ /g, '_');
  const url = `${BASE_URL}/beers?beer_name=${fixedTerm}`;

  return fetch(url)
    .then(returnJson)
    .then(returnBody);
};
