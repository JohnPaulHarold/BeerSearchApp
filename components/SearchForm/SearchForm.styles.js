import { StyleSheet } from 'react-native';

export const SearchFormStyles = StyleSheet.create({});
