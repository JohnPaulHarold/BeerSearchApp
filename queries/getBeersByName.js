import gql from 'graphql-tag';

export const GET_BEERS_BY_NAME = gql`
  query GetBeersByName($name: String!) {
    beers(name: $name) {
      id
      name
      image_url
    }
  }
`;
