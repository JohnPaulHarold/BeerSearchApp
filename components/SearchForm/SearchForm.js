/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React, { useState } from 'react';
import { AppRegistry } from 'react-native';
import { Form, Item, Label, Input, Button, Text, Container } from 'native-base';
import { SearchResults } from '../SearchResults/SearchResults';
import { ApolloConsumer } from 'react-apollo';
import { GET_BEERS_BY_NAME } from '../../queries/getBeersByName';

export const SearchForm = () => {
  const [searchTerm, setSearchTerm] = useState('');
  const [results, setResults] = useState(null);

  return (
    <ApolloConsumer>
      {client => (
        <Container>
          <Form>
            <Item floatingLabel>
              <Label>Search for beers</Label>
              <Input testID="text-input" onChangeText={setSearchTerm} />
            </Item>
            <Button
              testID="search-btn"
              disabled={searchTerm === ''}
              onPress={async () => {
                try {
                  const { data } = await client.query({
                    query: GET_BEERS_BY_NAME,
                    variables: { name: searchTerm }
                  });

                  setResults(data.beers);
                } catch (e) {
                  console.error(e.message);
                }
              }}
            >
              <Text>Search</Text>
            </Button>
          </Form>
          <SearchResults results={results} />
        </Container>
      )}
    </ApolloConsumer>
  );
};

AppRegistry.registerComponent('BeerSearchApp', () => SearchForm);
