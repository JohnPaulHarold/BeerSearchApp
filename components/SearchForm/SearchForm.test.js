/**
 * @format
 */
jest.useFakeTimers();

import 'react-native';
import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { Button } from 'native-base';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent, act } from 'react-native-testing-library';
import { SearchForm } from './SearchForm';
import { searchBeersByName } from '../../api/beer';
import { GET_BEERS_BY_NAME } from '../../queries/getBeersByName';
import { get } from 'https';

jest.mock('../../api/beer');

const request = {
  query: GET_BEERS_BY_NAME,
  variables: {
    name: 'peace'
  }
};

const mockSuccess = [
  {
    request,
    result: {
      data: {
        beers: [
          {
            id: '73',
            name: 'Black Eyed King Imp - Vietnamese Coffee Edition',
            image_url: 'https://images.punkapi.com/v2/73.png',
            __typename: 'Beer'
          },
          {
            id: '82',
            name: 'Hopped-Up Brown Ale - Prototype Challenge',
            image_url: 'https://images.punkapi.com/v2/82.png',
            __typename: 'Beer'
          },
          {
            id: '102',
            name: 'Peach Therapy',
            image_url: 'https://images.punkapi.com/v2/keg.png',
            __typename: 'Beer'
          },
          {
            id: '166',
            name: 'Vagabond Pale ALe - Prototype Challenge',
            image_url: 'https://images.punkapi.com/v2/166.png',
            __typename: 'Beer'
          },
          {
            id: '179',
            name: 'How To Disappear Completely',
            image_url: 'https://images.punkapi.com/v2/179.png',
            __typename: 'Beer'
          },
          {
            id: '274',
            name: 'Prototype Black Rye IPA',
            image_url: null,
            __typename: 'Beer'
          }
        ]
      }
    }
  }
];

const mockError = [
  {
    request,
    error: new Error('Something went wrong')
  }
];

const mockSearchTerm = 'peace';

describe('SearchForm', () => {
  it('renders correctly', () => {
    const render = renderer.create(
      <MockedProvider mocks={mockSuccess} addTypename={false}>
        <SearchForm />
      </MockedProvider>
    );
    expect(render).toBeTruthy();
    expect(render).toMatchSnapshot();
  });

  it('handles search', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={mockSuccess} addTypename={false}>
        <SearchForm />
      </MockedProvider>
    );

    const textEl = getByTestId('text-input');
    const btnEl = getByTestId('search-btn');

    fireEvent.changeText(textEl, mockSearchTerm);
    fireEvent.press(btnEl);
  });

  it('handles error', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={mockError} addTypename={false}>
        <SearchForm />
      </MockedProvider>
    );

    const textEl = getByTestId('text-input');
    const btnEl = getByTestId('search-btn');

    fireEvent.changeText(textEl, mockSearchTerm);
    fireEvent.press(btnEl);
  });
});
