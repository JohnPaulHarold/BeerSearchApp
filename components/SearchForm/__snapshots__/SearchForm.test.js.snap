// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRadius": undefined,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "paddingTop": 5,
              "top": 6,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 17,
              }
            }
          >
            Search for beers
          </Text>
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "marginLeft": null,
            "paddingBottom": 7,
            "paddingLeft": null,
            "paddingRight": 5,
            "paddingTop": 3,
            "top": 10,
          }
        }
        testID="text-input"
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#b5b5b5",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
      testID="search-btn"
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        Search
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      }
    }
  />
</View>
`;
